request-manager/graph/grapher.go

```diff
@@ -287,30 +310,29 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 		}
 	}
 
-	// Append sink node to ordering
 	seqGraph.Order = append(seqGraph.Order, seqGraph.Sink)
```

Comment:
Another confusing part of the graph logic: in `newSeqGraph` the sink node is created, so why do we have to add it here? I really think `Graph` is leaking way too much logic.